File: aro-drive-driver-vite/src/firebase/walletService.js
import { db } from './config';
import { 
  collection, addDoc, serverTimestamp, doc, onSnapshot 
} from 'firebase/firestore';

export const requestTopup = async (driverId, driverName, amount, method) => {
  try {
    const docRef = await addDoc(collection(db, "topup_requests"), {
      driverId,
      driverName,
      amount: Number(amount),
      method,
      status: 'pending',
      createdAt: serverTimestamp(),
      updatedAt: serverTimestamp()
    });
    return docRef.id;
  } catch (error) {
    console.error("Error requesting topup:", error);
    throw error;
  }
};

export const observeDriverBalance = (driverId, callback) => {
  return onSnapshot(doc(db, "drivers", driverId), (doc) => {
    if (doc.exists()) {
      callback(doc.data().balance || 0);
    }
  });
};

File: aro-driver-kotlin/app/src/main/java/com/arodriverkotlin/service/WalletService.kt
package com.arodriverkotlin.service

import com.google.firebase.firestore.FieldValue
import com.google.firebase.firestore.FirebaseFirestore
import kotlinx.coroutines.tasks.await

object WalletService {
    private val db = FirebaseFirestore.getInstance()

    suspend fun requestTopup(uid: String, name: String, amount: Long, method: String = "manual") {
        db.collection("topup_requests").add(
            mapOf(
                "driverId" to uid,
                "driverName" to name,
                "amount" to amount,
                "method" to method,
                "status" to "pending",
                "createdAt" to FieldValue.serverTimestamp(),
                "updatedAt" to FieldValue.serverTimestamp(),
            )
        ).await()
    }
}
